src/ServiceControl.Transports.Msmq/CheckDeadLetterQueue.cs

```diff
@@ -13,31 +14,96 @@ public class CheckDeadLetterQueue : CustomCheck
         {
             Logger.Debug("MSMQ Dead Letter Queue custom check starting");
 
-            dlqPerformanceCounter = new PerformanceCounter(
-                categoryName: "MSMQ Queue",
-                counterName: "Messages in Queue",
-                instanceName: "Computer Queues",
-                readOnly: true);
+            categoryName = Read("Msmq/PerformanceCounterCategoryName", "MSMQ Queue");
+            counterName = Read("Msmq/PerformanceCounterName", "Messages in Queue");
+            counterInstanceName = Read("Msmq/PerformanceCounterInstanceName", "Computer Queues");
+
+            try
+            {
+                dlqPerformanceCounter = new PerformanceCounter(categoryName, counterName, counterInstanceName, readOnly: true);
+            }
+            catch (InvalidOperationException ex)
+            {
+                Logger.Error(CounterMightBeLocalized(categoryName, counterName, counterInstanceName), ex);
+            }
         }
 
         public override Task<CheckResult> PerformCheck()
         {
             Logger.Debug("Checking Dead Letter Queue length");
-            var currentValue = dlqPerformanceCounter.NextValue();
+            float currentValue;
+            string result;
+            try
+            {
+                if (dlqPerformanceCounter == null)
+                {
+                    throw new InvalidOperationException("Unable to create performance counter instance.");
+                }
+
+                currentValue = dlqPerformanceCounter.NextValue();
+            }
+            catch (InvalidOperationException ex)
+            {
+                result = CounterMightBeLocalized(categoryName, counterName, counterInstanceName);
+                Logger.Warn(result, ex);
+                return CheckResult.Failed(result);
+            }
 
             if (currentValue <= 0)
             {
                 Logger.Debug("No messages in Dead Letter Queue");
                 return CheckResult.Pass;
             }
 
-            var result = $"{currentValue} messages in the Dead Letter Queue on {Environment.MachineName}. This could indicate a problem with ServiceControl's retries. Please submit a support ticket to Particular using support@particular.net if you would like help from our engineers to ensure no message loss while resolving these dead letter messages.";
+            result = $"{currentValue} messages in the Dead Letter Queue on {Environment.MachineName}. This could indicate a problem with ServiceControl's retries. Please submit a support ticket to Particular using support@particular.net if you would like help from our engineers to ensure no message loss while resolving these dead letter messages.";
```

Comment:
Is it worth pulling this out into a method like the `ContainerMightBeLocalized()`? Only for readability.